### DISCUSSION TITLE
Query string encoded square brackets

### DISCUSSION THREAD

**Discussion by dorinniscu** [livewire/livewire#3030] [7. Bugs (archived)]
I am using the "query string" feature for filters.

Everything works as expected for int / string properties but the array square brackets are encoded.

Example:
`/products?sort=name&direction=asc&perPage=48&color%5B0%5D=1&color%5B1%5D=2`

If I use nested filters, the result becomes much confusing.

Expected result: 
`/products?sort=name&direction=asc&perPage=48&color[]=1&color[]=2`

Technically works fine. It is only an aesthetic issue.

**Comment by calebporzio** [comment]
I'm sure we just url encode this data. Are square brackets safe in query strings?

**Comment by FaridAghili** [comment]
Makes sense to me, square brackets shouldn't get encoded as it makes the URL confusing and not user friendly.

@calebporzio 
Square brackets are safe, having square brackets in URLs (for filtering some attribute based on multiple values, etc.) is a very common approach.

**Comment by nullthoughts** [comment]
I was able to address this by decoding the url: https://github.com/livewire/livewire/commit/778735d311c21821b26403b10c7ba876fa30f9db

### Before
`?filters%5Bmake%5D%5B0%5D=Tesla&filters%5Bmake%5D%5B1%5D=VW`

### After
`?filters[make][0]=Tesla&filters[make][1]=VW`

More info (and query string customization): https://github.com/livewire/livewire/discussions/2036#discussioncomment-1168681
